contracts/zero-ex/contracts/src/features/LimitOrdersFeature.sol

```diff
@@ -0,0 +1,875 @@
+        }
+
+        // Check for expiration.
+        if (expiry <= uint64(block.timestamp)) {
+            orderInfo.status = LibLimitOrder.OrderStatus.EXPIRED;
+            return;
+        }
+
+        // Check if the order was cancelled by salt.
+        if (stor.makerToMakerTokenToTakerTokenToMinValidOrderSalt
+                [maker]
+                [address(makerToken)]
+                [address(takerToken)] > salt)
+        {
+            orderInfo.status = LibLimitOrder.OrderStatus.CANCELLED;
+            return;
+        }
+        orderInfo.status = LibLimitOrder.OrderStatus.FILLABLE;
+    }
+
+    /// @dev Cancel a limit or RFQ order directly by its order hash.
+    /// @param orderHash The order's order hash.
+    function _cancelOrderHash(bytes32 orderHash)
+        private
+    {
+        LibLimitOrdersStorage.Storage storage stor =
+            LibLimitOrdersStorage.getStorage();
+        // Set the high bit on the raw taker token fill amount to indicate
+        // a cancel. It's OK to cancel twice.
+        stor.orderHashToTakerTokenFilledAmount[orderHash] =
+            stor.orderHashToTakerTokenFilledAmount[orderHash] | HIGH_BIT;
+
+        emit OrderCancelled(orderHash);
+    }
+
+    /// @dev Fill a limit order. Private variant. Does not refund protocol fees.
+    /// @param params Function params.
+    /// @return ethProtocolFeePaid How much protocol fee was paid.
+    /// @return takerTokenFilledAmount How much maker token was filled.
+    /// @return makerTokenFilledAmount How much maker token was filled.
+    function _fillLimitOrderPrivate(FillLimitOrderPrivateParams memory params)
+        private
+        returns (
+            uint256 ethProtocolFeePaid,
+            uint128 takerTokenFilledAmount,
+            uint128 makerTokenFilledAmount
+        )
+    {
+        LibLimitOrder.OrderInfo memory orderInfo = getLimitOrderInfo(params.order);
+
+        // Must be fillable.
+        if (orderInfo.status != LibLimitOrder.OrderStatus.FILLABLE) {
+            LibLimitOrdersRichErrors.OrderNotFillableError(
+                orderInfo.orderHash,
+                uint8(orderInfo.status)
+            ).rrevert();
+        }
+
+        // Must be fillable by the taker.
+        if (params.order.taker != address(0) && params.order.taker != params.taker) {
+            LibLimitOrdersRichErrors.OrderNotFillableByTakerError(
+                orderInfo.orderHash,
+                msg.sender,
+                params.order.taker
+            ).rrevert();
+        }
+
+        // Must be fillable by the sender.
+        if (params.order.sender != address(0) && params.order.sender != params.sender) {
+            LibLimitOrdersRichErrors.OrderNotFillableBySenderError(
+                orderInfo.orderHash,
+                msg.sender,
+                params.order.sender
+            ).rrevert();
+        }
+
+        // Signature must be valid for the order.
+        {
+            address signer = LibSignature.getSignerOfHash(
```

Comment:
Does `LibSignature.getSignerOfHash` revert if the `signer` is `address(0)`?

Reply:
ye